test/dlc_tests/sanity/test_dlc_labels.py

```diff
@@ -42,4 +45,100 @@ def test_dlc_major_version_label(image, region):
+    processor = test_utils.get_processor_from_image_uri(image)
+
+    # Assign a string of numbers associated with python version in tag. Python major version is not sufficient to
+    # define DLC major version
+    python_major_minor_version = re.search(r'-py(\d{2,})', image).group(1)
+
+    root_dir = os.path.join(dlc_dir, framework, job_type, "docker")
+
+    # Skip older FW versions that did not use this versioning scheme
+    references = {"tensorflow2": "2.2.0", "tensorflow1": "1.16.0", "mxnet": "1.7.0", "pytorch": "1.5.0"}
+    if test_utils.is_tf1(image):
+        reference_fw = "tensorflow1"
+    elif test_utils.is_tf2(image):
+        reference_fw = "tensorflow2"
+    else:
+        reference_fw = framework
+    if processor != "eia" and packaging.version.parse(fw_version) < packaging.version.parse(references[reference_fw]):
+        pytest.skip(
+            f"Not enforcing new versioning scheme on old image {image}. "
+            f"Started enforcing version scheme on the following: {references}"
+        )
+
+    # Find all Dockerfile.<processor> for this framework/job_type's Major.Minor version
+    dockerfiles = []
+    fw_version_major_minor = re.match(r"(\d+\.\d+)", fw_version).group(1)
+    for root, dirnames, filenames in os.walk(root_dir):
+        for filename in filenames:
+            if filename == f"Dockerfile.{processor}":
+                dockerfile_path = os.path.join(root_dir, root, filename)
+                if "example" not in dockerfile_path and f"{os.sep}{fw_version_major_minor}" in dockerfile_path:
+                    dockerfiles.append(dockerfile_path)
+
+    # For the collected dockerfiles above, note the DLC major versions in each Dockerfile if python version matches
+    # the current image under test
+    versions = {}
+    dlc_label_regex = re.compile(r'LABEL dlc_major_version="(\d+)"')
+    python_version_regex = re.compile(r'ARG PYTHON_VERSION=(\d+\.\d+)')
+    for dockerfile in dockerfiles:
+        with open(dockerfile, "r") as df:
+            dlc_version = None
+            python_version = None
+            for line in df:
+                major_version_match = dlc_label_regex.match(line)
+                python_version_match = python_version_regex.match(line)
+                if major_version_match:
+                    dlc_version = int(major_version_match.group(1))
+                elif python_version_match:
+                    python_version = python_version_match.group(1).replace('.', '')
+
+            # Raise errors if dlc major version label and python version arg are not found in Dockerfile
+            if not dlc_version:
+                raise DLCMajorVersionLabelNotFound(f"Cannot find dlc_major_version label in {dockerfile}")
+            if not python_version:
+                raise DLCPythonVersionNotFound(f"Cannot find PYTHON_VERSION arg in {dockerfile}")
+            if python_version == python_major_minor_version:
```

Comment:
What if we add the support for a certain python version after several release? e.g. we add support for py38 for images, and py38 image DLC versions is not beginning with version 1.0. Do we also need to address this case especially once it emerges?